src/core/ext/filters/client_channel/subchannel.cc

```diff
@@ -952,11 +957,28 @@ void Subchannel::OnConnectingFinishedLocked(grpc_error_handle error) {
             time_until_next_attempt.millis());
     SetConnectivityStateLocked(GRPC_CHANNEL_TRANSIENT_FAILURE,
                                grpc_error_to_absl_status(error));
+    // Ensure grpc remains initialized when the timer callback runs. The
+    // subchannel could get deleted when the timer callback is run and
+    // subchannel deletion may destroye some pollset_sets which required grpc
+    // to be initialized.
+    grpc_init();
```

Comment:
Note that adding grpc init and shutdown to the subchannel code has been tried in the past but had to be reverted.  I don't know what the exact problem with this was in the past, but I think this is the right solution.  So I suggest running a TGP on this change to see if there are any problems, and if there are, we should figure out how to address them to make this work properly.

Reply:
On the other hand, I just remembered we deleted C#, I'll update it

Reply:
TGP passed for this PR (cl/453263832). I think we can give it a shot and land it